docs/core-concepts/components.md

```diff
@@ -0,0 +1,113 @@
+4. Execution Controller: watches Work objects. When Work objects are created, the controller will distribute the resources to member clusters.
+
+
+
+### karmada-scheduler
+
+This component is responsible for scheduling k8s native API resource objects (including CRD resources) to member clusters, including scheduling strategies and timing. At the same time, it also supports extending scheduling capabilities through plugins.
+
+For some resource objects that need to calculate the replica ratio, the scheduling plugin will first filter out some clusters, and then make a decision on the number of replicas that the resource object should distribute in each cluster.
+
+
+
+### karmada-webhook
+
+This component is responsible for verifying the format of resources object which is in the charge of , as well as intercepting and modifying relevant data before it entering the ETCD.
+
+
+
+### etcd
+
+Consistent and highly-available key value store used as Karmada' backing store for all karmada API objects. 
+
+
+
+## Data Plane Components
+
+
+
+### karmada-agent
+
+Karmada has two manage mode such as Push and Pull, karmada-agent shall be deployed on each Pull mode member cluster, responsible for actively pulling workload from the control plane, and applying it to current member cluster. Besides, the agent also in charge of collecting and reporting current cluster status to control plane.
+
+
+
+## Addons
+
+
+
+### karmada-scheduler-estimator
+
+This component is responsible for providing the karmada-scheduler with a more precise desired state of the running instances of the member cluster. Early karmada-scheduler only supported scheduling the number of replicas based on the total number of cluster resources. In this case, scheduling failure occurred when the total cluster resources were sufficient but each node resources were insufficient.
+
+To address this issue, the estimator component was introduced, which calculates the number of callable replicas for each node based on resource requests, thereby calculating the true number of schedulable replicas for a cluster.
+
+
+
+### karmada-descheduler
+
+This component is a cronjob responsible for detecting all Deployments at regular intervals (every 2 minutes by default), and triggering reschedule based on instance state changes in member clusters. The descheduler only takes effect when the scheduling strategy is dynamic division and it perceives how many instance state has changed by calling scheduler-estimator.
+
+
+
+### karmada-search
+
+Karmada search is used to cache resource objects and events deployed in member clusters, and to provide retrieval services externally through search APIs.
+
+Unlike the query services provided by the aggregation API, cache retrieval does not require accessing the target cluster and is targeted at scenarios that require frequent data search and analysis, such as building real-time resources and event dashboards.
+
+
+
+### karmadactl / kubectl-karmada
```

Comment:
It might be better to publicly introduce these two tasks separately, as they have different usage methods from the perspective of user experience.